inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
```

Comment:
Of course you'd have to define the meaning of the offsets slightly differently:

offset `n` | followed by
---|---
0 - 2 | `n` data bytes and a 3
4 - 254 | `n - 1` data bytes and a 3
255 | `n - 1` data bytes